src/libsystemd/sd-bus/bus-socket.c

```diff
@@ -183,22 +191,39 @@ static int bus_socket_auth_verify_client(sd_bus *b) {
                 start = e + 2;
         }
 
-        /* Nice! We got all the lines we need. First check the OK
-         * line */
+        /* Nice! We got all the lines we need. First check the DATA line. */
+
+        if (d - (char*) b->rbuffer == 4) {
+                if (memcmp(b->rbuffer, "DATA", 4))
```

Comment:
nitpick: we usually avoid relying on C's downgrade-to-bool feature unless something is actually boolean or boolean-like, or is a pointer. This here isn't either really, so weed to an explicit `memcmp() != 0`